app/scripts/controllers/network/provider-api-tests/createClient.test.js

```diff
@@ -0,0 +1,180 @@
+        // currently
+        'eth_getBlockTransactionCountByNumber',
+        'eth_getTransactionByBlockNumberAndIndex',
+
+        'eth_getBlockTransactionCountByHash',
+        'eth_getFilterLogs',
+        'eth_getTransactionByBlockHashAndIndex',
+        'eth_getUncleByBlockHashAndIndex',
+        'eth_getUncleByBlockNumberAndIndex',
+        'eth_getUncleCountByBlockHash',
+        'eth_getUncleCountByBlockNumber',
+      ];
+      assumingNoBlockParam.forEach((name) =>
+        describe(`method name: ${name}`, () => {
+          testsForRpcMethodAssumingNoBlockParam(name, { providerType });
+        }),
+      );
+    });
+
+    describe('methods with blockhashes in their result', () => {
+      const methodsWithBlockHashInResponse = [
+        'eth_getTransactionByHash',
+        'eth_getTransactionReceipt',
+      ];
+      methodsWithBlockHashInResponse.forEach((method) => {
+        describe(`method name: ${method}`, () => {
+          testsForRpcMethodsThatCheckForBlockHashInResponse(method);
+        });
+      });
+    });
+
+    describe('eth_chainId', () => {
+      it('does not hit custom provider, instead returning the chain id that maps to the custom providers network, as a hex string', async () => {
+        const chainId = await withClient(
+          { network: 'goerli', type: providerType, chainId: '0x5' },
+          ({ makeRpcCall }) => {
+            return makeRpcCall({
+              method: 'eth_chainId',
+            });
+          },
+        );
+
+        expect(chainId).toStrictEqual('0x5');
+      });
+    });
+  });
+
+  describe('not included in eth json-rpc spec', () => {
+    describe('methods not handled by middleware', () => {
+      const notHandledByMiddleware = [
+        { name: 'custom_rpc_method', numberOfParameters: 1 },
+        { name: 'eth_subscribe', numberOfParameters: 1 },
+        { name: 'eth_unsubscribe', numberOfParameters: 1 },
+        { name: 'net_listening', numberOfParameters: 0 },
+        { name: 'net_peerCount', numberOfParameters: 0 },
+        { name: 'parity_nextNonce', numberOfParameters: 1 },
+      ];
+      notHandledByMiddleware.forEach(({ name, numberOfParameters }) => {
+        describe(`method name: ${name}`, () => {
+          testsForRpcMethodNotHandledByMiddleware(name, {
+            numberOfParameters,
+            providerType,
+          });
+        });
+      });
+    });
+
+    describe('methods that assume there is no block param', () => {
+      const assumingNoBlockParam = [
+        'eth_protocolVersion',
+        'web3_clientVersion',
+      ];
+      assumingNoBlockParam.forEach((name) =>
+        describe(`method name: ${name}`, () => {
+          testsForRpcMethodAssumingNoBlockParam(name, { providerType });
+        }),
+      );
+    });
+
+    it(`net_version does not hit ${providerType} provider, instead returning the networkId that maps to the rpc network, as a decimal string`, async () => {
```

Comment:
Nit: What do you think about using a `describe` to hold tests for `net_version`, even if there is only one test? This way the names of the tests (when viewed by running this one test file on its own) are consistent, and the test file in general follows a general shape.

Reply:
These tests have been moved to `createInfuraClient.test.js`.